yb-voyager/cmd/common.go

```diff
@@ -526,3 +527,67 @@ func getDefaultPGSchema(schema string) (string, bool) {
 		return "", true
 	}
 }
+
+func CleanupChildProcesses() {
+	log.Info("Cleaning up child processes...")
+	myPid := os.Getpid()
+	processes, err := ps.Processes()
+	if err != nil {
+		log.Errorf("failed to read processes: %v", err)
+	}
+	childProcesses := lo.Filter(processes, func(process ps.Process, _ int) bool {
+		return process.PPid() == myPid
+	})
+	for _, process := range childProcesses {
+		pid := process.Pid()
+		log.Infof("shutting down child pid %d", pid)
+		err := ShutdownProcess(pid, 60)
+		if err != nil {
+			log.Errorf("shut down child pid %d : %v", pid, err)
+		} else {
+			log.Infof("shut down child pid %d", pid)
+		}
+	}
+}
+
+// this function wait for process to exit after signalling it to stop
+func ShutdownProcess(pid int, forceShutdownAfterSeconds int) error {
+	err := signalProcess(pid, syscall.SIGTERM)
+	if err != nil {
+		return fmt.Errorf("send sigterm to %d: %v", pid, err)
+	}
+	waitForProcessToExit(pid, forceShutdownAfterSeconds)
+	return nil
+}
+
+func signalProcess(pid int, signal syscall.Signal) error {
+	process, _ := os.FindProcess(pid) // Always succeeds on Unix systems
+	log.Infof("sending signal=%s to process with PID=%d", signal.String(), pid)
+	err := process.Signal(signal)
```

Comment:
What if the process ends between `FindProcess()` and `process.Signal()`?

Reply:
This function will error out, but the caller `CleanupChildProccesses` is an exit handler, it won't return an error/ panic, it will simply log the error and continue.